Sources/DeepLearning/Layer.swift

```diff
@@ -36,12 +36,16 @@ public protocol Layer: Differentiable & KeyPathIterable
     func applied(to input: Input) -> Output
 }
 
-public extension Layer {
+public extension Layer
+    where TangentVector : AdditiveArithmetic, CotangentVector : AdditiveArithmetic,
```

Comment:
To make this more error-prone, let's add these constraints to all differential operators defined in [AutoDiff.swift](https://github.com/apple/swift/blob/tensorflow/stdlib/public/core/AutoDiff.swift) and add FIXME's explaining that they are workarounds for [SR-9595](https://bugs.swift.org/browse/SR-9595).

Reply:
Adding constraints to differential operators is a very good idea.